plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven.test/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenArchetypeProjectWizardTest.java

```diff
@@ -79,19 +90,40 @@ public void testArchetypeDefaultSelection() {
 
   @Test
   public void testSuggestPackageName() {
-    Assert.assertEquals("aa.bb",
-        MavenAppEngineStandardWizardPage.suggestPackageName("aa.bb"));
+    assertEquals("aa.bb", MavenAppEngineStandardWizardPage.suggestPackageName("aa.bb"));
 
-    Assert.assertEquals("aA.Bb",
-        MavenAppEngineStandardWizardPage.suggestPackageName("aA.Bb"));
+    assertEquals("aA.Bb", MavenAppEngineStandardWizardPage.suggestPackageName("aA.Bb"));
 
-    Assert.assertEquals("aa.bb",
+    assertEquals("aa.bb",
         MavenAppEngineStandardWizardPage.suggestPackageName(" a  a\t . b\r b \n"));
 
-    Assert.assertEquals("aa.bb",
+    assertEquals("aa.bb",
         MavenAppEngineStandardWizardPage.suggestPackageName("....aa....bb..."));
 
-    Assert.assertEquals("aa._01234bb", MavenAppEngineStandardWizardPage.suggestPackageName(
+    assertEquals("aa._01234bb", MavenAppEngineStandardWizardPage.suggestPackageName(
         "aa`~!@#$%^&*()-+=[]{}<>\\|:;'\",?/._01234bb"));
   }
+
+  @Test
+  public void testAutoPackageNameSetterOnGroupIdChange_whitespaceInGroupId() {
+    wizard.setContainer(mock(IWizardContainer.class));
+    wizard.createPageControls(shell);
+    MavenAppEngineStandardWizardPage page =
+        (MavenAppEngineStandardWizardPage) wizard.getPage("basicNewProjectPage");
+
+    page.groupIdField.setText(" ");  // setText() triggers VerifyEvent.
+    assertEquals("", page.javaPackageField.getText());
+
+    page.groupIdField.setText(" a");
+    assertEquals("a", page.javaPackageField.getText());
+
+    page.groupIdField.setText(" a ");
+    assertEquals("a", page.javaPackageField.getText());
+
+    page.groupIdField.setText(" a a");
```

Comment:
The current behavior is to try its best to suggest a valid package name by removing, e.g., invalid characters. Since users may go back and forth between a valid group ID state and an invalid state while typing, I think this behavior is acceptable. What else do you think it should do if the group ID is incorrect?